src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -307,75 +320,76 @@ public static void ValidateRestPathForRelationalDbs(RuntimeConfig runtimeConfig)
-                    statusCode: HttpStatusCode.ServiceUnavailable,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+                exceptionMessageSuffix = "cannot be null or empty.";
             }
-
-            // A valid api path prefix should start with a forward slash '/'.
-            if (!apiPath.StartsWith("/"))
+            // A valid URI component should start with a forward slash '/'.
+            else if (!uriComponent.StartsWith("/"))
             {
-                throw new DataApiBuilderException(
-                    message: $"{apiType} path should start with a '/'.",
-                    statusCode: HttpStatusCode.ServiceUnavailable,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+                exceptionMessageSuffix = "should start with a '/'.";
+            }
+            else
+            {
+                uriComponent = uriComponent.Substring(1);
+                // URI component should not contain any reserved characters.
+                if (DoesUriComponentContainInvalidChars(uriComponent))
+                {
+                    exceptionMessageSuffix = URI_COMPONENT_WITH_RESERVED_CHARS_ERR_MSG;
+                }
             }
 
-            apiPath = apiPath[1..];
-
-            // API path prefix should not contain any reserved characters.
-            DoApiPathInvalidCharCheck(apiPath, apiType);
+            return exceptionMessageSuffix;
         }
 
         /// <summary>
-        /// Method to validate that the REST/GraphQL path prefix does not contain
-        /// any forbidden characters.
+        /// Method to validate that the REST/GraphQL API's URI component does not contain
+        /// any reserved characters.
         /// </summary>
-        /// <param name="apiPath">path prefix for rest/graphql apis</param>
-        /// <param name="apiType">Either REST or GraphQL</param>
-        /// <exception cref="DataApiBuilderException"></exception>
-        public static void DoApiPathInvalidCharCheck(string apiPath, ApiType apiType)
+        /// <param name="uriComponent">path prefix for rest/graphql apis</param>
+        public static bool DoesUriComponentContainInvalidChars(string uriComponent)
         {
-            if (_invalidApiPathCharsRgx.IsMatch(apiPath))
-            {
-                string errorMessage = INVALID_GRAPHQL_PATH_WITH_RESERVED_CHAR_ERR_MSG;
-                if (apiType == ApiType.REST)
-                {
-                    errorMessage = INVALID_REST_PATH_WITH_RESERVED_CHAR_ERR_MSG;
-                }
-
-                throw new DataApiBuilderException(
-                    message: errorMessage,
-                    statusCode: HttpStatusCode.ServiceUnavailable,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
-            }
+            return _invalidApiPathCharsRgx.IsMatch(uriComponent);
```

Comment:
now that Uri validation is generic, should the name `_invalidApiPathCharsRgx` be more generic?